src/display.rs

```diff
@@ -538,6 +538,51 @@ where
     f.pad_integral(true, "0x", encoded)
 }
 
+/// Given a `T:` [`fmt::Write`], `HexWriter` implements [`std::io::Write`]
+/// and writes the source bytes to its inner `T` as hex characters.
+#[cfg(any(test, feature = "std"))]
+#[cfg_attr(docsrs, doc(cfg(any(test, feature = "std"))))]
+pub struct HexWriter<T>(T, Case);
+
+#[cfg(any(test, feature = "std"))]
+#[cfg_attr(docsrs, doc(cfg(any(test, feature = "std"))))]
+impl<T> HexWriter<T> {
+    /// Creates a `HexWriter` that writes the source bytes to `dest` as hex characters
+    /// in the given `case`.
+    pub fn new(dest: T, case: Case) -> Self { Self(dest, case) }
+    /// Consumes this `HexWriter` returning the inner `T`.
+    pub fn into_inner(self) -> T { self.0 }
+}
+
+#[cfg(any(test, feature = "std"))]
+#[cfg_attr(docsrs, doc(cfg(any(test, feature = "std"))))]
+impl<T> std::io::Write for HexWriter<T>
+where
+    T: core::fmt::Write,
+{
+    /// # Errors
+    ///
+    /// If no bytes could be written to this `HexWriter`, and the provided buffer is not empty,
+    /// returns [`std::io::ErrorKind::Other`], otherwise returns `Ok`.
+    fn write(&mut self, buf: &[u8]) -> Result<usize, std::io::Error> {
+        let mut n = 0;
+        let table = self.1.table();
+        for byte in buf {
+            let pair = table.byte_to_hex(*byte);
+            if self.0.write_str(&pair).is_err() {
+                break;
+            }
+            n += 1;
+        }
+        if n == 0 && !buf.is_empty() {
+            Err(std::io::ErrorKind::Other.into())
+        } else {
+            Ok(n)
+        }
+    }
+    fn flush(&mut self) -> Result<(), std::io::Error> { Ok(()) }
+}
```

Comment:
Note that if we implement byte-by-byte writing we really should override `write_all` to provide a more efficient implementation.